ops/terraform/modules/resources/s3_pii/templates/iam-writer-policy.json

```diff
@@ -4,23 +4,43 @@ ${jsonencode(
   "Id": "bfd-${env}-${name}-iam-writer-policy",
   "Statement": [
     {
-      "Sid": "BFDPIIS3ReadAll",
+      "Sid": "BFDPIIS3ListBuckets",
       "Effect": "Allow",
       "Action": [
         "s3:ListAllMyBuckets",
         "s3:HeadBucket"
       ],
-      "Resource": "*"
+      "Resource": "arn:aws:s3:::*"
+    },
+    {
+      "Sid": "BFDPIIS3ListBucketObjects",
+      "Effect": "Allow",
+      "Action": [
+        "s3:ListBucket",
+      ],
+      "Resource": [
+        "arn:aws:s3:::${bucket_id}"
+      ]
+    },
+    {
+      "Sid": "BFDPIIS3ReadAll",
+      "Effect": "Allow",
+      "Action": [
+        "s3:GetObject",
+        "s3:GetObjectVersion"
+      ],
+      "Resource": [
+        "arn:aws:s3:::${bucket_id}",
+        "arn:aws:s3:::${bucket_id}/*"
+      ]
     },
     {
       "Sid": "BFDPIIS3WriteAssume",
       "Effect": "Allow",
       "Action": [
-        "s3:ListBucket*",
+        "s3:ListBucketMultiPartUploads",
         "s3:ListMultipartUploadParts",
-        "s3:GetObject*",
```

Comment:
Looks like you're turning off quite a few things here by disabling `GetObject*`, most of which look harmless enough and some of which might be used: GetObjectAttributes, GetObjectRetention, GetObjectTagging, GetObjectVersion, GetObjectVersionAttributes, GetObjectVersionTagging. Similarly with DeleteObjectVersion.

I _think_ this is probably fine, and I'm more than willing to take your word on it if there aren't being used. Just wanted to make sure it was purposeful.

Reply:
I made this more explicit. We can add future actions as needed. Thanks!